File: examples/java/Advancing_front_reconstruction_example.java
import CGAL.Kernel.Point_3;
import CGAL.Polyhedron_3.Polyhedron_3;
import CGAL.Point_set_3.Point_set_3;
import CGAL.Advancing_front_surface_reconstruction.CGAL_Advancing_front_surface_reconstruction;

import java.util.Vector;

public class Advancing_front_reconstruction_example{
  public static void main(String arg[]){

    Point_set_3 points = new Point_set_3("../data/oni.xyz");
    if (points.empty())
    {
      System.out.println("Error: cannot read file ../data/oni.xyz");
      return;
    }

    Polyhedron_3 P = new Polyhedron_3();

    CGAL_Advancing_front_surface_reconstruction.
      advancing_front_surface_reconstruction(points, P);

    P.write_to_file("oni.off");
  }
}

File: examples/python/Advancing_front_reconstruction_example.py
from __future__ import print_function
from sys import exit
from sys import stderr

from CGAL.CGAL_Kernel import Point_3
from CGAL.CGAL_Polyhedron_3 import Polyhedron_3
from CGAL.CGAL_Point_set_3 import Point_set_3
from CGAL.CGAL_Advancing_front_surface_reconstruction import *


points = Point_set_3("../data/oni.xyz")
if points.empty():
  stderr.write("Error: cannot read file ../data/oni.xyz\n")
  exit()

P = Polyhedron_3()

advancing_front_surface_reconstruction(points, P);

P.write_to_file("oni.off");

